src/fancy-tools/wwwcRegionTool.js

```diff
@@ -2,102 +2,33 @@
 import toolColors from './../stateManagement/toolColors.js';
 
-const isEmptyObject = (obj) => Object.keys(obj).length === 0 && obj.constructor === Object;
-
-/** Calculates the minimum, maximum, and mean value in the given pixel array */
-const calculateMinMaxMean = (storedPixelLuminanceData, globalMin, globalMax) => {
-  const numPixels = storedPixelLuminanceData.length;
-  let min = globalMax;
-  let max = globalMin;
-  let sum = 0;
-
-  if (numPixels < 2) {
-    return {
-      min,
-      max,
-      mean: (globalMin + globalMax) / 2
-    };
-  }
-
-  for (let index = 0; index < numPixels; index++) {
-    const spv = storedPixelLuminanceData[index];
-
-    min = Math.min(min, spv);
-    max = Math.max(max, spv);
-    sum += spv;
-  }
-
-  return {
-    min,
-    max,
-    mean: sum / numPixels
-  };
-};
-
-/** Calculates the minimum and maximum value in the given pixel array */
-const applyWWWCRegion = (evt, config) => {
-  const eventData = evt.detail;
-  const { image, element } = eventData;
-  const { start: startPoint, end: endPoint } = evt.detail.handles;
-
-  // Get the rectangular region defined by the handles
-  let left = Math.min(startPoint.x, endPoint.x);
-  let top = Math.min(startPoint.y, endPoint.y);
-  let width = Math.abs(startPoint.x - endPoint.x);
-  let height = Math.abs(startPoint.y - endPoint.y);
-
-  // Bound the rectangle so we don't get undefined pixels
-  left = clip(left, 0, image.width);
-  top = clip(top, 0, image.height);
-  width = Math.floor(Math.min(width, Math.abs(image.width - left)));
-  height = Math.floor(Math.min(height, Math.abs(image.height - top)));
-
-  // Get the pixel data in the rectangular region
-  const pixelLuminanceData = getLuminance(element, left, top, width, height);
-
-  // Calculate the minimum and maximum pixel values
-  const minMaxMean = calculateMinMaxMean(pixelLuminanceData, image.minPixelValue, image.maxPixelValue);
-
-  // Adjust the viewport window width and center based on the calculated values
-  const viewport = eventData.viewport;
-
-  if (config.minWindowWidth === undefined) {
-    config.minWindowWidth = 10;
-  }
-
-  viewport.voi.windowWidth = Math.max(Math.abs(minMaxMean.max - minMaxMean.min), config.minWindowWidth);
-  viewport.voi.windowCenter = minMaxMean.mean;
-
-  external.cornerstone.setViewport(element, viewport);
-  external.cornerstone.updateImage(element);
-};
-
 export default class extends baseTool {
   constructor (name) {
-    const strategies = {
-      applyWWWCRegion
-    };
-
     super({
       name: name || 'wwwcRegion',
-      strategies,
       supportedInteractionTypes: ['mouse', 'touch'],
       configuration: {
         minWindowWidth: 10
       }
     });
 
-    this._mouseUpCallback = this._mouseUpCallback.bind(this);
-    this._mouseMoveCallback = this._mouseMoveCallback.bind(this);
-
     this._resetHandles();
+    // Touch
+    this.activeTouchStartCallback = this._startOutliningRegion.bind(this);
```

Comment:
I'm in two minds about this. Obviously we are just starting to flesh out all the tools with touch interfaces. I wonder if its best to have everything in one class, like here. Or two classes e.g wwwcMouseTool and wwwcTouchTool. This particular case isn't too heavy, but I'm wondering if it makes the class difficult to parse by eye. Not sure 🤷‍♂️ .

Reply:
Yeah, I think this is still better currently. I'm not too hot on touch controls, is there effectively a 1:1 analogue with e.g. drag/click, or is the touch functionality more bespoke than that? If their is an analogue we could probably get rid of a load of boilerplate with some clever interfacing and inheritance.

Reply:
@JamesAPetts, I think there's definitely a path for some consolidation. I'm hoping as well wrap up the remaining "baked in tools" that the path becomes a little clearer.